db/merge_operator.cc

```diff
@@ -23,6 +27,84 @@ bool MergeOperator::FullMergeV2(const MergeOperationInput& merge_in,
                    &merge_out->new_value, merge_in.logger);
 }
 
+bool MergeOperator::FullMergeV3(const MergeOperationInputV3& merge_in,
+                                MergeOperationOutputV3* merge_out) const {
+  assert(merge_out);
+
+  MergeOperationInput in_v2(merge_in.key, nullptr, merge_in.operand_list,
+                            merge_in.logger);
+
+  std::string new_value;
+  Slice existing_operand(nullptr, 0);
+  MergeOperationOutput out_v2(new_value, existing_operand);
+
+  return std::visit(
+      overload{
+          [&](const auto& existing) -> bool {
+            using T = std::decay_t<decltype(existing)>;
+
+            if constexpr (std::is_same_v<T, Slice>) {
+              in_v2.existing_value = &existing;
+            }
+
+            const bool result = FullMergeV2(in_v2, &out_v2);
+            if (!result) {
+              merge_out->op_failure_scope = out_v2.op_failure_scope;
+              return false;
+            }
+
+            if (existing_operand.data()) {
+              merge_out->new_value = existing_operand;
+            } else {
+              merge_out->new_value = std::move(new_value);
+            }
+
+            return true;
+          },
+          [&](const WideColumns& existing_columns) -> bool {
+            const bool has_default_column =
```

Comment:
minor: I'm wondering if we want to make this check available as a static function in WideColumnsHelper (or other better place). We have this same check in a few different places.

https://github.com/facebook/rocksdb/blob/main/db/merge_helper.cc#L137C1-L138

https://github.com/facebook/rocksdb/blob/main/db_stress_tool/no_batched_ops_stress.cc#L168-L169

https://github.com/facebook/rocksdb/blob/main/db_stress_tool/no_batched_ops_stress.cc#L254-L255

Reply:
Good call! Let me take care of that as a separate PR.